File: cinema-backend/src/main/java/org/example/cinemabackend/cinema/application/dto/request/CreateScreeningRoomRequest.java
package org.example.cinemabackend.cinema.application.dto.request;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;
import lombok.Builder;
import org.example.cinemabackend.cinema.application.dto.response.ProjectionTechnologyNameResponse;

import java.util.Set;

@Builder
public record CreateScreeningRoomRequest(
        @NotBlank @Size(max = 50) String name,
        @NotNull CreatSeatRequest[][] seats,
        @NotNull Set<@NotNull ProjectionTechnologyNameResponse> supportedTechnologies
) {
}

File: cinema-frontend/src/app/cinema/dtos/request/create-screening-room.request.ts
import {
  ProjectionTechnologyResponse
} from "../../../projection-technology/dtos/response/projection-technology.response";
import {CreateSeatRequest} from "./create-seat.request";

export class CreateScreeningRoomRequest {
  constructor(
    readonly name: string,
    readonly seats: CreateSeatRequest[][],
    readonly supportedTechnologies: ProjectionTechnologyResponse[],
  ) {
  }
}
